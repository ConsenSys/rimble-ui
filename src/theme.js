import { tint, shade } from 'polished';

const success = '#28C081';
const danger = '#DC2C10';
const warning = '#FD9D28';
const info = '#36ADF1';

// theme.js
export default {
  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64],
  fontWeights: [0, 300, 400, 600, 700],
  letterSpacings: [0, 1, 2, 4, 8],
  lineHeights: {
    solid: 1,
    title: 1.25,
    copy: 1.5,
  },
  fonts: {
    serif: 'athelas, georgia, times, serif',
    sansSerif: '"Source Sans Pro", -apple-system, sans-serif',
  },
  space: [0, 4, 8, 16, 32, 64, 128, 256],
  radii: ['0', '4px', '8px', '16px'],
  width: [0, 16, 32, 64, 128, 256],
  minWidths: [0, 16, 32, 64, 128, 256],
  maxWidths: [0, 16, 32, 64, 128, 256, 512, 768, 1024, 1536],
  heights: [0, 16, 32, 64, 128, 256],
  minHeights: [0, 16, 32, 64, 128, 256],
  maxHeights: [0, 16, 32, 64, 128, 256],
  borders: [0, '1px solid transparent'],
  borderWidths: ['0', '1px', '2px', '4px'],
  shadows: [
    '0',
    '0px 2px 4px rgba(0, 0, 0, 0.1)',
    '0 7px 14px rgba(50,50,93,.1)',
  ],
  opacity: {
    disabled: 0.4,
  },
  colors: {
    primary: '#4E3FCE',
    blue: '#007ce0',
    navy: '#004175',
    copyColor: '#3F3D4B',
    // black: '#000',
    black: '#000e1a',
    'near-black': '#111',
    'dark-gray': '#333',
    'mid-gray': '#555',
    // gray: ' #777',
    grey: '#CCC',
    silver: '#999',
    'light-silver': '#aaa',
    'moon-gray': '#ccc',
    'light-gray': '#eee',
    'near-white': '#f4f4f4',
    white: '#fff',
    transparent: 'transparent',
    blacks: [
      'rgba(0,0,0,.0125)',
      'rgba(0,0,0,.025)',
      'rgba(0,0,0,.05)',
      'rgba(0,0,0,.1)',
      'rgba(0,0,0,.2)',
      'rgba(0,0,0,.3)',
      'rgba(0,0,0,.4)',
      'rgba(0,0,0,.5)',
      'rgba(0,0,0,.6)',
      'rgba(0,0,0,.7)',
      'rgba(0,0,0,.8)',
      'rgba(0,0,0,.9)',
    ],
    whites: [
      'rgba(255,255,255,.0125)',
      'rgba(255,255,255,.025)',
      'rgba(255,255,255,.05)',
      'rgba(255,255,255,.1)',
      'rgba(255,255,255,.2)',
      'rgba(255,255,255,.3)',
      'rgba(255,255,255,.4)',
      'rgba(255,255,255,.5)',
      'rgba(255,255,255,.6)',
      'rgba(255,255,255,.7)',
      'rgba(255,255,255,.8)',
      'rgba(255,255,255,.9)',
    ],
    danger: danger,
    warning: warning,
    success: success,
    info: info,
  },
  zIndices: [0, 9, 99, 999, 9999],
  messageStyle: {
    base: {
      color: shade(0.4, '#AAA'),
      backgroundColor: tint(0.9, '#AAA'),
      borderColor: '#AAA',
    },
    success: {
      color: shade(0.4, success),
      backgroundColor: tint(0.9, success),
      borderColor: success,
    },
    warning: {
      color: shade(0.4, warning),
      backgroundColor: tint(0.9, warning),
      borderColor: warning,
    },
    danger: {
      color: shade(0.4, danger),
      backgroundColor: tint(0.9, danger),
      borderColor: danger,
    },
    info: {
      color: shade(0.4, info),
      backgroundColor: tint(0.9, info),
      borderColor: info,
    },
  },
};
