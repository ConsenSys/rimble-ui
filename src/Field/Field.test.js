import Field from "./";
import ThemeProvider from "../ThemeProvider";
import theme from "../theme";

describe("Field component sanity", () => {
  it("has name", () => {
    expect(Field.displayName).toBe("Field");
  });

  it("matches default snapshot", () => {
    const component = render(<Field />);
    expect(component).toMatchSnapshot();
  });

  it("matches themed snapshot", () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Field />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
