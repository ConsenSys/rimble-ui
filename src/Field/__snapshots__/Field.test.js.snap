// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component sanity matches default snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  text-align: left;
}

.c0 {
  box-sizing: border-box;
}

.c1 {
  margin-bottom: 8px;
  color: #3F3D4B;
  font-size: 14px;
  text-align: left;
  font-size: 14px;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-weight: 600;
  text-align: left;
  line-height: 1.5;
}

.c1 {
  box-sizing: border-box;
}

<label
  class="c0"
  display="flex"
>
  <div
    class="c1"
    color="copyColor"
    font-family="sansSerif"
    font-size="1"
    font-weight="3"
  >
    
  </div>
</label>
`;

exports[`Field component sanity matches themed snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  text-align: left;
}

.c1 {
  box-sizing: border-box;
}

.c2 {
  margin-bottom: 8px;
  color: #3F3D4B;
  font-size: 14px;
  text-align: left;
  font-size: 14px;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-weight: 600;
  text-align: left;
  line-height: 1.5;
}

.c2 {
  box-sizing: border-box;
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <label
    class="c1"
    display="flex"
  >
    <div
      class="c2"
      color="copyColor"
      font-family="sansSerif"
      font-size="1"
      font-weight="3"
    >
      
    </div>
  </label>
</div>
`;
